test/apm_registry.js

```diff
@@ -61,21 +60,30 @@ contract('APMRegistry', accounts => {
     })
 
     it('fails to create empty repo name', async () => {
-        return assertRevert(async () => {
-            await registry.newRepo('', repoDev, { from: apmOwner })
-        })
+        try {
```

Comment:
Can we still run all of this inside a function like `assertRevert(...)` does? 

We can expect that the function provided returns a Promise to a receipt, and after that we can check it:

```
async function assertThrows(codeBlock, message, errorCode) {
    try {
        const result = await codeBlock()
        assert.equal(result.receipt.status, 0, 'should have failed status')
    } catch (e) {
        return assertError(e, errorCode, message)
    }
    assert.fail('should have thrown before')
}
```

Reply:
I tried that at first but the `result` that was returned was always undefined. That's why I went to using the try-catch blocks whenever the function was called in the tests. I will mess around with trying to see if I can get that logic reduced into a function.